File: kuakua_agent/schemas/summary.py
from pydantic import BaseModel, Field


class AppUsage(BaseModel):
    name: str = Field(min_length=1)
    duration: float = Field(ge=0)
    category: str = Field(default="other")


class SummaryResponse(BaseModel):
    date: str
    total_hours: float = Field(ge=0)
    work_hours: float = Field(ge=0)
    entertainment_hours: float = Field(ge=0)
    other_hours: float = Field(ge=0)
    top_apps: list[AppUsage] = Field(default_factory=list)
    focus_score: int = Field(ge=0, le=100)
    praise_text: str
    suggestions: list[str] = Field(default_factory=list)

File: electron/src/renderer/store/summary.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { getTodaySummary, getSummary } from '@/api'

export interface SummaryData {
  date: string
  total_hours: number
  work_hours: number
  entertainment_hours: number
  other_hours: number
  top_apps: Array<{ name: string; duration: number; category: string }>
  focus_score: number
  praise_text: string
  suggestions: string[]
}

export const useSummaryStore = defineStore('summary', () => {
  const summary = ref<SummaryData | null>(null)
  const loading = ref(false)
  const error = ref<string | null>(null)

  async function fetchTodaySummary() {
    loading.value = true
    error.value = null
    try {
      const response = await getTodaySummary()
      if (response.data.status === 'success' && response.data.data) {
        summary.value = response.data.data
      } else {
        error.value = response.data.message || '获取总结失败'
      }
    } catch (e: unknown) {
      error.value = e instanceof Error ? e.message : '网络错误'
    } finally {
      loading.value = false
    }
  }

  async function fetchSummary(date: string) {
    loading.value = true
    error.value = null
    try {
      const response = await getSummary(date)
      if (response.data.status === 'success' && response.data.data) {
        summary.value = response.data.data
      } else {
        error.value = response.data.message || '获取总结失败'
      }
    } catch (e: unknown) {
      error.value = e instanceof Error ? e.message : '网络错误'
    } finally {
      loading.value = false
    }
  }

  return { summary, loading, error, fetchTodaySummary, fetchSummary }
})